### PR TITLE
Update system-contracts.mdx

### PR DESCRIPTION
**Pull request by cpengilly**
add l2 predeploy contract address table

### CODE REVIEW

**Comment by sbvegan** [comment]
fixes: https://github.com/ethereum-optimism/docs/issues/114

File: pages/dapp-developers/contracts/system-contracts.mdx [hunk 1]
```diff
@@ -13,8 +13,39 @@ In this guide, we'll show you how to work with OP Mainnet contracts directly fro
 ## Find contract addresses
 
 You'll need to find the address of the particular contract that you want to interact with before you can actually interact with it.
-Check out the [Networks and Connection Details page](/op-networks) for links to the contract addresses for each network.
-You can also find the addresses for all networks in the [deployments folder](https://github.com/ethereum-optimism/optimism/tree/master/packages/contracts/deployments) of the [`contracts` package](https://github.com/ethereum-optimism/optimism/tree/master/packages/contracts).
+- Check out the [Networks and Connection Details page](/op-networks) for links to the contract addresses for each network.
+- You can also find the addresses for all networks in the [deployments folder](https://github.com/ethereum-optimism/optimism/tree/master/packages/contracts/deployments) of the [`contracts` package](https://github.com/ethereum-optimism/optimism/tree/master/packages/contracts).
```
Comment (sbvegan) [inline pages/dapp-developers/contracts/system-contracts.mdx:17]: These links are broken, they should go to the respective folders in here: https://github.com/ethereum-optimism/optimism/tree/master/packages/contracts-bedrock

File: pages/dapp-developers/contracts/system-contracts.mdx [hunk 2]
```diff
@@ -13,8 +13,39 @@ In this guide, we'll show you how to work with OP Mainnet contracts directly fro
...
+
+### L2 Predeploy Contract Addresses
+
+The following table includes each of the predeploys. The system version
+indicates when the predeploy was introduced. The possible values are `Legacy`
+or `Bedrock` or `Canyon`. Deprecated contracts should not be used.
+
+| Name                          | Address                                    | Introduced | Deprecated | Proxied |
+| ----------------------------- | ------------------------------------------ | ---------- | ---------- |---------|
+| LegacyMessagePasser           | 0x4200000000000000000000000000000000000000 | Legacy     | Yes        | Yes     |
```
Comment (sbvegan) [inline pages/dapp-developers/contracts/system-contracts.mdx:28]: I think the contract addresses would be nice as links to etherscan, for example:
```
[0x4200000000000000000000000000000000000000](https://optimistic.etherscan.io/address/0x4200000000000000000000000000000000000000)
```

File: pages/dapp-developers/contracts/system-contracts.mdx (hunk 1 above)
Comment (cpengilly) [reply pages/dapp-developers/contracts/system-contracts.mdx]: updated

File: pages/dapp-developers/contracts/system-contracts.mdx (hunk 2 above)
Comment (cpengilly) [reply pages/dapp-developers/contracts/system-contracts.mdx]: agreed!

**Review by sbvegan — APPROVED** [review]

**Pull request merged**